packages/dual-channel/README.md

```diff
@@ -0,0 +1,144 @@
+# Dual-channel
+
+Dual Channel is a React Component to simultaneously present paragraphs and images in the browser viewport.
+
+# Developer Document
+
+## Install
+
+Install dependencies:
+
+```bash
+make check-dep
+```
+
+## Dev
+
+```bash
+make dev-server
+```
+
+## Build
+
+```bash
+make build
+```
+
+### Test Embedded Code with Mock Data
+
+```bash
+make test-embedded-code
+```
+
+## Import React Component
+
+You can import either
+
+Option 1:
+
+```
+import dualChannel from '@twreporter/dual-channel'
+
+const ReactComponent = dualChannel.ReactComponent
+```
+
+or
+
+Option 2:
+
+```
+import ReactComponent from '@twreporter/dual-channel/lib/app'
+```
+
+If you use webpack to bundle,
+I suggest that you can use Option 2 to prevent bundle errors.
```

Comment:
Is there any further information about what kind of error will occur with option 1? And If option 1 is not reliable, do we still have to provide the interface by this way?

Reply:
The error comes from `googleapis` npm package.
In `src/sheets/index.js`, we `import {google} from 'googleapis'`.
However, `googleapis` uses some nodejs modules which could not be bundled by webpack.

Wouldn't `@twreporter/timeline` encounter the same problem?

Reply:
I use the option 2 to import the module when using webpack to prevent taking unused modules into bundle so I didn't found this problem. I think we can:

1. Keep current code, and add above info you mentioned into `README.md`

OR

2. Remove `googleapis` related helpers from package entry since it is not universal js on both client and server.

Reply:
I can add above information into  `README.md`

For 2., we need to remove `import Sheets from './sheets'` from `src/index.js`.
Therefore, if clients want to use `Sheets`, they need to `import Sheets from '@twreporter/dual-channel/lib/sheets'` by themselves.

Either way, clients needs to `import ReactComponent from '@twreporter/dual-channel/lib/app'` if they want to build webpack or  `import Sheets from '@twreporter/dual-channle/lib/sheets'` if they want to use `Sheets` API.

In my opinion, I will adopt Option 1. (add more information in `README.md`), and expose all of them in package entry.
How do you think?

Reply:
Both ways are good to me 👍 Option 2 are not so intuitive with `from '@twreporter/dual-channle/lib/sheets'` so it's just an acceptable way. It may be semantically better if the api is designed as something like `import { Sheets } from '@twreporter/dual-channel/server'` but I think this will lead to too many code changes.

Reply:
I found Option 3 to solve this problem.
In client's `webpack.config.js`, add `target: node` (see https://webpack.js.org/concepts/targets/#usage) to prevent from bundling nodejs built-in modules.
I will add Option 1 and Option 3 in `README.md`.

Reply:
> I found Option 3 to solve this problem.
> In client's `webpack.config.js`, add `target: node` (see https://webpack.js.org/concepts/targets/#usage) to prevent from bundling nodejs built-in modules.
> I will add Option 1 and Option 3 in `README.md`.

Option 3 is not working.
See the latest change in this PR https://github.com/twreporter/orangutan-monorepo/pull/22 .